### PR TITLE
add logger module

### PR DESCRIPTION
**Pull request by MaigeWhite**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: configs/rec/crnn/crnn_icdar15.yaml [hunk 1]
```diff
@@ -69,6 +69,7 @@ loss_scaler:
 
 train:
   ckpt_save_dir: './tmp_rec'
+  log_interval: 100
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_icdar15.yaml:72]: log_interval参数放到system下。

File: mindocr/utils/callbacks.py [hunk 2]
```diff
@@ -110,8 +112,12 @@ class EvalSaveCallback(Callback):
         loader (Dataset): dataloader
         saving_config (dict):
     """
+
     def __init__(self,
                  network,
+                 epoch_size,
```
Comment (SamitHuang) [inline mindocr/utils/callbacks.py:118]: 不用传epoch_size，这个callback内部可通过cb_params.epoch_num获取
另外，命名保持一致: num_epochs

File: tools/train.py [hunk 3]
```diff
@@ -112,37 +119,40 @@ def main(cfg):
...
+        logger,
+        loader_eval,
+        postprocessor=postprocessor,
+        metrics=[metric],
+        rank_id=rank_id,
+        ckpt_save_dir=cfg.train.ckpt_save_dir,
+        main_indicator=cfg.metric.main_indicator,
+        val_interval=cfg.system.get('val_interval', 1),
+        val_start_epoch=cfg.system.get('val_start_epoch', 1),
+        log_interval=cfg.train.log_interval,
```
Comment (SamitHuang) [inline tools/train.py:134]: log_interval 加一个默认值 ，cfg.system.get('log_interval', 100)

File: configs/rec/crnn/crnn_resnet34.yaml [hunk 4]
```diff
@@ -69,6 +69,7 @@ loss_scaler:
 
 train:
   ckpt_save_dir: './tmp_rec'
+  log_interval: 100
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_resnet34.yaml:72]: 删去，100为默认值

File: configs/rec/crnn/crnn_vgg7.yaml [hunk 5]
```diff
@@ -69,6 +69,7 @@ loss_scaler:
 
 train:
   ckpt_save_dir: './tmp_rec'
+  log_interval: 100
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_vgg7.yaml:72]: del

File: mindocr/utils/callbacks.py [hunk 6]
```diff
@@ -110,8 +112,12 @@ class EvalSaveCallback(Callback):
...
+                 batch_size,
+                 logger,
```
Comment (SamitHuang) [inline mindocr/utils/callbacks.py:120]: batch_size和logger这两个参数放在后面（按重要性和必要性排）

File: tools/train.py
Comment (SamitHuang) [inline tools/train.py:1]: 请添加logger的单元测试代码至tests/ut中。

File: mindocr/utils/callbacks.py (hunk 6 above)
Comment (MaigeWhite) [reply mindocr/utils/callbacks.py:118]: logger和batch_size是必须传的参数, 训练日志用logger输出，计算fps需要batch_size

File: configs/rec/crnn/crnn_resnet34.yaml (hunk 4 above)
Comment (MaigeWhite) [reply configs/rec/crnn/crnn_resnet34.yaml]: 检测数据比较小，设置为100 step_end日志信息就没法输出

File: tools/train.py [hunk 7]
```diff
@@ -117,42 +124,39 @@ def main(cfg):
...
-            f'Num steps per epoch: {num_batches}'
-            )
-        if 'name' in cfg.model:
-            print(f'Model: {cfg.model.name}')
-        else:
-            print(f'Model: {cfg.model.backbone.name}-{cfg.model.neck.name}-{cfg.model.head.name}')
-        print('='*40)
+    logger.info('=' * 40)
+    logger.info(
+        f'Num devices: {device_num if device_num is not None else 1}\n'
```
Comment (SamitHuang) [inline tools/train.py:144]: 这部分log信息有缺失，请与最新的commit同步。

File: configs/rec/crnn/crnn_icdar15.yaml [hunk 8]
```diff
@@ -3,6 +3,7 @@ system:
   distribute: False 
   amp_level: 'O0'
   seed: 42
+  log_interval: 100
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_icdar15.yaml:6]: 与默认值相同，可删去。

File: configs/rec/crnn/crnn_resnet34.yaml [hunk 9]
```diff
@@ -3,6 +3,7 @@ system:
   distribute: True 
   amp_level: 'O3'
   seed: 42
+  log_interval: 100
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_resnet34.yaml:6]: 与默认值相同，可删去。

File: configs/rec/crnn/crnn_vgg7.yaml [hunk 10]
```diff
@@ -3,6 +3,7 @@ system:
   distribute: True 
   amp_level: 'O3'
   seed: 42
+  log_interval: 100
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_vgg7.yaml:6]: 与默认值相同，可删去。

File: configs/det/dbnet/db_r50_synthtext.yaml [hunk 11]
```diff
@@ -3,6 +3,7 @@ system:
   distribute: True
   amp_level: 'O0'
   seed: 42
+  log_interval: 100
```
Comment (SamitHuang) [inline configs/det/dbnet/db_r50_synthtext.yaml:6]: 与默认值相同，可删去。

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**